File: runtime/flutter/services/textInputConnection.ts
import { IFontWeight } from "../../dart/ui/fontWeight";
import { IRect } from "../../dart/ui/rect";
import { ISize } from "../../dart/ui/size";
import { TextAlign } from "../../dart/ui/textAlign";
import { TextDirection } from "../../dart/ui/textDirection";
import { IMatrix4 } from "../../vector_math/matrix4";
import { ITextEditingValue } from "./textEditingValue";
import { ITextInputConfiguration } from "./textInputConfiguration";
declare const flutter: {
    services: {
        textInputConnectionDebugResetId: (props: { to: number }) => void;
    };
};
export interface ITextInputConnection {
    getAttached: () => boolean;
    show: () => void;
    requestAutofill: () => void;
    updateConfig: (configuration: ITextInputConfiguration) => void;
    setEditingState: (value: ITextEditingValue) => void;
    setEditableSizeAndTransform: (
        editableBoxSize: ISize,
        transform: IMatrix4
    ) => void;
    setComposingRect: (rect: IRect) => void;
    setCaretRect: (rect: IRect) => void;
    setStyle: (props: {
        fontFamily?: string | undefined;
        fontSize?: number | undefined;
        fontWeight?: IFontWeight | undefined;
        textAlign: TextAlign;
        textDirection: TextDirection;
    }) => void;
    close: () => void;
    connectionClosedReceived: () => void;
}
export class TextInputConnection {
    public static debugResetId(props: { to?: number }): void {
        return flutter.services.textInputConnectionDebugResetId({
            ...debugResetIdDefaultProps,
            ...props,
        });
    }
    private readonly _dart_getAttached: () => boolean = undefined as any;
    private readonly _dart_show: () => void = undefined as any;
    private readonly _dart_requestAutofill: () => void = undefined as any;
    private readonly _dart_updateConfig: (
        configuration: ITextInputConfiguration
    ) => void = undefined as any;
    private readonly _dart_setEditingState: (value: ITextEditingValue) => void =
        undefined as any;
    private readonly _dart_setEditableSizeAndTransform: (
        editableBoxSize: ISize,
        transform: IMatrix4
    ) => void = undefined as any;
    private readonly _dart_setComposingRect: (rect: IRect) => void =
        undefined as any;
    private readonly _dart_setCaretRect: (rect: IRect) => void =
        undefined as any;
    private readonly _dart_setStyle: (props: {
        fontFamily?: string | undefined;
        fontSize?: number | undefined;
        fontWeight?: IFontWeight | undefined;
        textAlign: TextAlign;
        textDirection: TextDirection;
    }) => void = undefined as any;
    private readonly _dart_close: () => void = undefined as any;
    private readonly _dart_connectionClosedReceived: () => void =
        undefined as any;
    public getAttached(): boolean {
        return this._dart_getAttached();
    }
    public show(): void {
        return this._dart_show();
    }
    public requestAutofill(): void {
        return this._dart_requestAutofill();
    }
    public updateConfig(configuration: ITextInputConfiguration): void {
        return this._dart_updateConfig(configuration);
    }
    public setEditingState(value: ITextEditingValue): void {
        return this._dart_setEditingState(value);
    }
    public setEditableSizeAndTransform(
        editableBoxSize: ISize,
        transform: IMatrix4
    ): void {
        return this._dart_setEditableSizeAndTransform(
            editableBoxSize,
            transform
        );
    }
    public setComposingRect(rect: IRect): void {
        return this._dart_setComposingRect(rect);
    }
    public setCaretRect(rect: IRect): void {
        return this._dart_setCaretRect(rect);
    }
    public setStyle(props: {
        fontFamily?: string | undefined;
        fontSize?: number | undefined;
        fontWeight?: IFontWeight | undefined;
        textAlign: TextAlign;
        textDirection: TextDirection;
    }): void {
        return this._dart_setStyle(props);
    }
    public close(): void {
        return this._dart_close();
    }
    public connectionClosedReceived(): void {
        return this._dart_connectionClosedReceived();
    }
}
const debugResetIdDefaultProps = {
    to: 1,
};

File: lib/cfr/builtins/libs/flutter/services/textInputConnection.dart
import 'dart:core' as _fac9;
import 'dart:ui' as _a643;

import 'package:flutter/src/services/text_input.dart' as _8577;

import 'package:vector_math/vector_math_64.dart' as _db98;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedTextInputConnection extends _36c2.VMManagedBox<_8577.TextInputConnection> {VMManagedTextInputConnection({required this.table, required this.vmObject, required this.hydroState}) : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) { table  [
'getAttached'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { return [vmObject.attached,]; } );
table  [
'show'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { vmObject.show();
return []; } );
table  [
'requestAutofill'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { vmObject.requestAutofill();
return []; } );
table  [
'updateConfig'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { vmObject.updateConfig(_36c2.maybeUnBoxAndBuildArgument<_8577.TextInputConfiguration, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState));
return []; } );
table  [
'setEditingState'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { vmObject.setEditingState(_36c2.maybeUnBoxAndBuildArgument<_8577.TextEditingValue, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState));
return []; } );
table  [
'setEditableSizeAndTransform'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 
vmObject.setEditableSizeAndTransform(_36c2.maybeUnBoxAndBuildArgument<_a643.Size, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState), _36c2.maybeUnBoxAndBuildArgument<_db98.Matrix4, _fac9.dynamic>(luaCallerArguments  [
2
], parentState: hydroState));
return []; } );
table  [
'setComposingRect'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { vmObject.setComposingRect(_36c2.maybeUnBoxAndBuildArgument<_a643.Rect, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState));
return []; } );
table  [
'setCaretRect'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { vmObject.setCaretRect(_36c2.maybeUnBoxAndBuildArgument<_a643.Rect, _fac9.dynamic>(luaCallerArguments  [
1
], parentState: hydroState));
return []; } );
table  [
'setStyle'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { 



vmObject.setStyle(fontFamily: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'fontFamily'
] : null, fontSize: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'fontSize'
] : null?.toDouble(), fontWeight: _36c2.maybeUnBoxAndBuildArgument<_a643.FontWeight?, _fac9.dynamic>(luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'fontWeight'
] : null, parentState: hydroState), textAlign: _36c2.maybeUnBoxEnum(values: _a643.TextAlign.values, boxedEnum: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'textAlign'
] : null), textDirection: _36c2.maybeUnBoxEnum(values: _a643.TextDirection.values, boxedEnum: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'textDirection'
] : null));
return []; } );
table  [
'close'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { vmObject.close();
return []; } );
table  [
'connectionClosedReceived'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { vmObject.connectionClosedReceived();
return []; } ); }

final _36c2.HydroTable table;

final _36c2.HydroState hydroState;

final _8577.TextInputConnection vmObject;

 }

void loadTextInputConnection({required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { 
table  [
'textInputConnectionDebugResetId'
] = _36c2.makeLuaDartFunc(func: (_fac9.List<_fac9.dynamic> luaCallerArguments) { _8577.TextInputConnection.debugResetId(to: luaCallerArguments.length >= 2 ? luaCallerArguments  [
1
]  [
'to'
] : null);
return []; } );
_36c2.registerBoxer<_8577.TextInputConnection>(boxer: ({required _8577.TextInputConnection vmObject, required _36c2.HydroState hydroState, required _36c2.HydroTable table}) { return  VMManagedTextInputConnection(vmObject: vmObject, hydroState: hydroState, table: table); } ); } 
